src/libs/actions/IOU.js

```diff
@@ -724,11 +721,327 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {Object} report
+ * @param {Number} amount
+ * @param {String} currency
+ * @param {String} comment
+ * @param {String} paymentMethodType
+ * @param {String} managerEmail - Email of the person sending the money
+ * @param {Object} recipient - The user receiving the money
+ * @returns {Object}
+ */
+function getSendMoneyParams(report, amount, currency, comment, paymentMethodType, managerEmail, recipient) {
+    const newIOUReportDetails = JSON.stringify({
+        amount,
+        currency,
+        requestorEmail: OptionsListUtils.addSMSDomainIfPhoneNumber(recipient.login),
+        comment,
+        idempotencyKey: Str.guid(),
+    });
+
+    const recipientEmail = OptionsListUtils.addSMSDomainIfPhoneNumber(recipient.login);
+    let chatReport = report.reportID ? report : null;
+    let isNewChat = false;
+    if (!chatReport) {
+        chatReport = ReportUtils.getChatByParticipants([recipientEmail]);
+    }
+    if (!chatReport) {
+        chatReport = ReportUtils.buildOptimisticChatReport([recipientEmail]);
+        isNewChat = true;
+    }
+    const optimisticIOUReport = ReportUtils.buildOptimisticIOUReport(recipientEmail, managerEmail, amount, chatReport.reportID, currency, preferredLocale, true);
+
+    // This will be deprecated soon, in case the migration happens before this PR is merged we'll need to adjust the code here
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+
+    const optimisticIOUReportAction = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount,
+        currency,
+        comment,
+        [recipient],
+        CONST.IOU.PAYMENT_TYPE.ELSEWHERE,
+        '',
```

Comment:
I see the problem is that we generate the iouReportID before for the iouReport, but what we could still do is to add here the new generatated IOUTransactionID instead of passing in empty string. Then if it is type PAY and has transactionID defined it is sending money.

Just unsure if we should keep adding more optional parameters here.

Reply:
You're looking at code from sendMoney here, in `getPayMoneyRequestParams` I don't see how can I get an `IOUTransactionID` it's only generated inside `buildOptimisticIOUReportAction `?

Reply:
Let me give this a try.